shared/enclave/server/index.ts

```diff
@@ -0,0 +1,68 @@
+import { protocol, App } from 'electron';
+import handlers from './handlers';
+import { PROTOCOL_NAME, isValidEventType } from 'shared/enclave/utils';
+import { EnclaveMethods, EnclaveMethodParams, EnclaveResponse } from 'shared/enclave/types';
+
+export function registerServer(app: App) {
+  // Register protocol scheme
+  protocol.registerStandardSchemes([PROTOCOL_NAME]);
+
+  app.on('ready', () => {
+    // Register custom protocol behavior
+    protocol.registerStringProtocol(PROTOCOL_NAME, async (req, cb) => {
+      let res: EnclaveResponse;
+
+      try {
+        const method = getMethod(req);
+        const params = getParams(method, req);
+        const data = await handlers[method](params);
+        res = { data };
+      } catch (err) {
+        console.error(`Request to '${req.url}' failed with error:`, err);
```

Comment:
Should stringify the error here incase it contains an object

Reply:
It's the second arg in console.error which does the stringifying for you, not a string interpolated thing.

Reply:
Wouldn't it print [Object object] in some cases?

Reply:
No, it's the same as doing
```
console.log({ my: 'object' });
```
or
```
console.log(new Error('test'));
```
it'll always print a "nice" version of it.